eng/common/TestResources/New-TestResources.ps1

```diff
@@ -176,6 +198,25 @@ if ($ProvisionerApplicationId) {
         }
     }
 }
+elseif (!$CI)
+{
+    # check if user is logged in and login into
+    $context = Get-AzContext;
+    if (!$context)
+    {
+        Log "You are not logged in, connecting to 'Azure SDK Developer Playground'"
+        Connect-AzAccount -Subscription $defaultSubscription
+    }
+    
+    # If no test application id is specified create a new service principal
+    if (!$TestApplicationId) {        
+        Log "TestApplicationId was not specified, creating a new service principal."
+        $servicePrincipal = New-AzADServicePrincipal -Role Owner
```

Comment:
Does that need to be removed at the end of the run, if it was dynamic?   _(or is it already and I'm just overlooking?)_

Reply:
Yeah, in theory, you are right. But we never told anyone to remove their service principals after manual creation so I decided to postpone fixing the removing part.

Reply:
Is there any downside to removing it at the end of the script?

Reply:
Yeah, tests won't run because they won't be able to sign in :)

Reply:
You could only remove it in the `Remove-TestResources` script, assuming the same principal is used to run the tests (by default they are). You could save that to an environment variable or some other means to know which to remove.

Reply:
Filed https://github.com/Azure/azure-sdk-tools/issues/1078